### PR TITLE
Fixes#827 Enabled Long click on GeoTrace and Shape widget

### LINKED ISSUE #827: Long press on button doesn't work using GeoTraceWidget/GeoShapeWidget
#### Software and hardware versions 
Collect v1.5.1

#### Problem description
We are not able to remove response using [GeoTraceWidget](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/widgets/GeoTraceWidget.java) and [GeoShapeWidget](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/widgets/GeoShapeWidget.java) when we long press on buttons.

#### Steps to reproduce the problem
1. Open the attached form.
2. Long press on button "Starte GeoTrace" or "Start GeoShape"

#### Expected behavior
2. "Remove response" option should be visible as with every other widget

[locations.txt](https://github.com/opendatakit/collect/files/895981/locations.txt)

### PR DESCRIPTION
**Pull request by rijuldhir**
Fixes #827

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/widgets/GeoShapeWidget.java [hunk 1]
```diff
@@ -221,6 +221,9 @@ public void setFocus(Context context) {
     @Override
     public void setOnLongClickListener(OnLongClickListener l) {
         // TODO Auto-generated method stub
+        createShapeButton.setOnLongClickListener(l);
+        mStringAnswer.setOnLongClickListener(l);
+        mAnswerDisplay.setOnLongClickListener(l);
 
     }
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/widgets/GeoShapeWidget.java]: Please remove "TODO" comment as there is nothing to do now and remove empty line"

File: collect_app/src/main/java/org/odk/collect/android/widgets/GeoTraceWidget.java [hunk 2]
```diff
@@ -213,6 +213,9 @@ public void setFocus(Context context) {
     @Override
     public void setOnLongClickListener(OnLongClickListener l) {
         // TODO Auto-generated method stub
+        createTraceButton.setOnLongClickListener(l);
+        mStringAnswer.setOnLongClickListener(l);
+        mAnswerDisplay.setOnLongClickListener(l);
 
 
     }
```
Comment (grzesiek2010) [inline collect_app/src/main/java/org/odk/collect/android/widgets/GeoTraceWidget.java]: The same as above.

**Comment by rijuldhir** [comment]
@grzesiek2010 done

**Comment by grzesiek2010** [comment]
Looks ok to me.

**Comment by nribeka** [comment]
@rijuldhir I think you need to branch off from your master branch and add your changes to that branch.

https://help.github.com/articles/about-pull-requests/

**Comment by lognaturel** [comment]
@nribeka @rijuldhir Agreed for next time but it's not a big deal for this one. It's mostly for the committer's convenience. @nribeka if you're trying to check this change out you can do `git checkout -b foo rijuldhir/master` or something like that.

**Comment by rijuldhir** [comment]
@nribeka Sure Will do it from next time after this one gets merged.

**Comment by lognaturel** [comment]
Great, @rijuldhir! We just need some feedback from @yanokwa since he mentioned maybe there was a reason not to do this.

**Comment by yanokwa** [comment]
I don't remember which side of the argument I was on, but clearing was not included because you already have an option to delete the data once you launch the Geo activity. There might be another reason, but I can't think of it. @jnordling Do you remember why we did it this way?

**Comment by lognaturel** [comment]
You didn't ask my opinion but I'll share it anyway! :) I like consistency! Even though the widgets have their own clearing mechanism, I think it's nice to have one action that works across all widgets.

**Comment by jnordling** [comment]
Ohh I see now.. Not in the widget. On the widget factory screen to open the widgets. I think this was over looked on our part. Im not against it. I think the consistency could be nice. The only other way to remove it would be to enter the widget and click the trash can then exit out. Right? @yanokwa I we just missed it all together.

**Pull request merged**